### ISSUE TITLE
Size of GUI cannot be changed

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#45]
Under certain (unknown) circumstances, the size and position of the GUI window cannot be changed.

The problem can be fixed by deleting the `runtime_config.yaml`, which is of course no solution for the issue.

@Rbelder provided his runtime config with which the issue occured:

```
LAST_PATH_OPEN_SAVE: /home/Rbelder/Desktop/cvafda
WINDOW_POS: !!python/tuple
- 0
- 0
WINDOW_SIZE: !!python/tuple
- 1920
- 1200
```

After deleting the runtime config, it was recreated and looked like this:

```
WINDOW_POS: !!python/tuple
- 60
- 150
WINDOW_SIZE: !!python/tuple
- 1642
- 981
```

I tried setting the values of the problematic config in my runtime config, but I couldn't reproduce the error.

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-02-12 08:23:56+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
So here some doubts, facts to reproduce the failure and my opinion.

Doubts:
The runtime config does not has a fullscreen boolean.

Facts:
- I have two different screens 1920x1200 and 1920x1080 -> means left is the bigger one and screens upper edges are the same -> so there is a cap at the lower edge of the both screens + the smaller screen has a task bar (reason for screen size hight 1033) and the bigger screen not.
  See the both full screen run_time_config.yaml below.
- If the bigger left screen is closed in full screen mode the setup is locked in fullscreen on the left screen after re-running.
- If the smaller right screen is closed in full screen mode the full screen button has to be pressed two times to reduce the window size (normal is one click) after re-running RAFCON.

Opinion:
- the run time config should be adapted to get a more general setting of the screen especially for changing window sizes. -> if the screen is in full screen that is an information which is may never covered completely by the window size -> lets introduce a full screen boolean (even the size and position and the real screen sizes are may checked in the code this will stay a source of problem)

So I think the cause of the problem is in the lines where it is checked for full screen mode.

LEFT SCREEN:

WINDOW_POS: !!python/tuple
- 0
- 0

WINDOW_SIZE: !!python/tuple
- 1920
- 1200

RIGHT SCREEN:

WINDOW_POS: !!python/tuple
- 1920
- 0

WINDOW_SIZE: !!python/tuple
- 1920
- 1033


Originally created by @Rbelder  at 2016-02-12 10:04:41+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
@franzlst We have talked about the issue lately and have struggled if it is still occurring and reproducible. So yes it is. :disappointed: 

As described above I create the failure by using 2 different sized screens. Closing on the second screen (bigger) and it opens on the smaller first screen with stoking in Full-screen.

As long as I can reproduce it I will take a look into it if I can fix it in a way as considered above or with checking the reasons for unwanted jump on the not focused screen and also not previously used screen.


Originally created by @Rbelder  at 2017-02-07 10:45:04+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
You could check if links and nodes has a similar problem, e.g. with process output windows. It also resizes windows according to the stored size: https://rmc-github-server/schm-fl/links_and_nodes/blob/master/manager/process_output.py


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2017-02-07 11:30:01+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
The failure can not be reproduced with "links and nodes" and i dread it is because of our special main window frame arrangement. I also also can reproduce it with the undocking feature.


Originally created by @Rbelder  at 2017-02-07 13:14:51+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
So the bug seems to be solved by a commit included in the pull request #286. Sorry that I put it in a feature pull request.

Both the feature and the bug fix are changing the runtime config usage and storing of gui meta data (size, position plus now maximized and undocked). Thats why I put it in one request. Both are improving also the size position recovery abilities of the GUI as I have seen.

@franzlst Please do a small check, you also know the undock feature and maybe you see a conflict I have not seen.


Originally created by @Rbelder  at 2017-02-07 17:10:24+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
`So here is the right place`

So I added a FAQ for this problem because it can not be fully avoided. The issue is closed because it is considered to be also strongly window manager and desktop depended. Further the performance of respective features is now strongly improved and the probability to reproduce it by accident is very low.

We can reopen the issue if somebody don't like to close it at this state. In this case I would create a "minor bug" or "none-critical-bug" label and assign it to this issue.


Originally created by @Rbelder  at 2017-02-08 15:56:30+00:00 (moved from RMC internal repository)
